HIRS_Utils/src/main/java/hirs/data/persist/tpm/PcrInfoShort.java

```diff
@@ -199,17 +203,33 @@ public final int hashCode() {
         return result;
     }
 
+
     /**
-     * Calculates the digest of the PCR values the same way a TPM computes the digest contained in
-     * the quote. Useful for TPM appraisal and for ensuring the digest of the collected PCR values
-     * match the digest in the quote.
+     * Calculates the SHA-1 digest of the PCR values the same way a TPM computes the digest
+     * contained in the quote. Useful for TPM appraisal and for ensuring the digest of the
+     * collected PCR values match the digest in the quote.
      *
      * @return byte array containing the digest
      * @throws NoSuchAlgorithmException
      *             if MessageDigest doesn't recognize "SHA-1"
      */
     public final byte[] getCalculatedDigest() throws NoSuchAlgorithmException {
-        MessageDigest messageDigest = MessageDigest.getInstance("SHA-1");
+        if (this.isTpm1()) {
+            return getCalculatedDigestTpmV1p2(MessageDigest.getInstance("SHA-1"));
+        } else {
+            return getCalculatedDigestTpmV2p0(MessageDigest.getInstance("SHA-256"));
+        }
+    }
+
+    /**
+     * Calculates the digest of the PCR values the same way a TPM computes the digest contained in
+     * the quote. Useful for TPM appraisal and for ensuring the digest of the collected PCR values
+     * match the digest in the quote.
+     *
+     * @param messageDigest message digest algorithm to use
+     * @return byte array containing the digest
+     */
+    public final byte[] getCalculatedDigestTpmV1p2(final MessageDigest messageDigest) {
```

Comment:
This seems like this could be a `private` method since the TPM 1.2 should only ever pass in "SHA-1" as a MessageDigest.

Reply:
I need to double check this, but I'm not sure that's actually possible. Several of the methods in this class are actually needed by the Appraiser to do the appraisal. Their use isn't restricted simply to within this class.

Reply:
Ah ok, I see your point. This is only referenced via getCalculatedDigest.

Reply:
fixed